scipy/stats/_continuous_distns.py

```diff
@@ -9309,6 +9309,205 @@ def _updated_ctor_param(self):
+            arg = [K, k, df, log_const]
+            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(_stats, cython_symbol, usr_data)
+
+            ranges = [(-np.inf, np.inf), (0, np.inf), (_a, _b)]
+            opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_moment, 3, 1)
+        return ufunc(K, k, df).astype("float64")
+
+    def _pdf(self, x, k, df):
+        cython_symbol = '_genstudentized_range_pdf'
+        _a, _b = self._get_support()
+        x = np.atleast_1d(x)
+
+        def _single_pdf(q, k, df):
+            log_const = _stats._genstudentized_range_pdf_logconst(k, df)
+            arg = [q, k, df, log_const]
+            usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+            llc = LowLevelCallable.from_cython(_stats, cython_symbol, usr_data)
+
+            ranges = [(-np.inf, np.inf), (_a, _b)]
+            opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+            return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_pdf, 3, 1)
+        return ufunc(x, k, df).astype("float64")
+
+    def _cdf(self, x, k, df):
+        _a, _b = self._get_support()
+        x = np.atleast_1d(x)
+
+        def _single_cdf(q, k, df):
+            # "When the degrees of freedom V are infinite the probability
+            # integral takes [on a] simpler form," and a single asymptotic
+            # integral is evaluated rather than the standard double integral.
+            # (Lund, Lund, page 205)
+            if df < 100000:
+                cython_symbol = '_genstudentized_range_cdf'
+                log_const = _stats._genstudentized_range_cdf_logconst(k, df)
+
+                arg = [q, k, df, log_const]
+                usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+                llc = LowLevelCallable.from_cython(
+                    _stats, cython_symbol, usr_data)
+
+                ranges = [(-np.inf, np.inf), (_a, _b)]
+                opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+            else:
+                cython_symbol = '_genstudentized_range_cdf_asymptotic'
+
+                arg = [q, k]
+                usr_data = np.array(arg, float).ctypes.data_as(ctypes.c_void_p)
+
+                llc = LowLevelCallable.from_cython(
+                    _stats, cython_symbol, usr_data)
+
+                ranges = [(-np.inf, _b)]
+                opts = dict(epsabs=1e-11, epsrel=1e-12)
+
+                return integrate.nquad(llc, ranges=ranges, opts=opts)[0]
+
+        ufunc = np.frompyfunc(_single_cdf, 3, 1)
+        return ufunc(x, k, df).astype("float64")
+
+
+studentized_range = studentized_range_gen(name='studentized_range', a=0,
+                                          b=np.inf)
```

Comment:
Can you try to figure out what `longname` parameters is used for and whether we should add it here?

Reply:
Probably shouldn't. From the `rv_continuous` doc 
>    This string is used as part of the first line of the docstring returned when a subclass has no docstring of its own. Note: 
>    longname exists for backwards compatibility, do not use for new subclasses.